insights/client/__init__.py

```diff
@@ -291,50 +298,51 @@ def update_rules(self):
             logger.debug("Bypassing rule update due to config "
                 "running in offline mode or auto updating turned off.")
         else:
-            return client.update_rules(self.config)
+            return client.update_rules(self.config, self.connection)
 
+    @_net
     def collect(self):
         # return collection results
-        tar_file = client.collect(self.config)
+        tar_file = client.collect(self.config, self.connection)
 
         # it is important to note that --to-stdout is utilized via the wrapper RPM
         # this file is received and then we invoke shutil.copyfileobj
         return tar_file
 
-    def register(self, force_register=False):
+    @_net
+    def register(self):
         """
-            returns (json): {'success': bool,
-                            'machine-id': uuid from API,
-                            'response': response from API,
-                            'code': http code}
+            returns (bool | None):
+                True - machine is registered
+                False - machine is unregistered
+                None - could not reach the API
         """
-        self.config.register = True
-        if force_register:
-            self.config.reregister = True
-        return client.handle_registration(self.config)
+        return client.handle_registration(self.config, self.connection)
 
+    @_net
     def unregister(self):
         """
             returns (bool): True success, False failure
         """
-        return client.handle_unregistration(self.config)
+        return client.handle_unregistration(self.config, self.connection)
 
+    @_net
     def get_registration_information(self):
         """
-            returns (json): {'machine-id': uuid from API,
-                            'response': response from API}
+            returns (json):
+                {'messages': [dotfile message, api message],
+                 'status': (bool) registered = true; unregistered = false
+                 'unreg_date': Date the machine was unregistered | None,
+                 'unreachable': API could not be reached}
         """
-        registration_status = client.get_registration_status(self.config)
-        return {'machine-id': client.get_machine_id(),
-                'registration_status': registration_status,
-                'is_registered': registration_status['status']}
+        return client.get_registration_status(self.config, self.connection)
 
     def upload(self, path, rotate_eggs=True):
```

Comment:
Shouldn’t this use the new __net_ decorator too? Even though it is not necessary, because some other method with it is always called before this one, it still uses and needs _self.connection_.

Reply:
agreed